src/main/java/filter/Parser.java

```diff
@@ -173,13 +174,25 @@ private FilterExpression parseQualifier(Token token) {
         qualifierName = qualifierName.substring(0, qualifierName.length() - 1).trim();
 
         QualifierType type = QualifierType.parse(qualifierName).orElse(QualifierType.FALSE);
-        return parseQualifierContent(type, false);
+        if (type == QualifierType.SORT) {
+            //special case for sort
```

Comment:
@dariusf I agree. For the distinction, are you referring to #1195? Btw, for the second issue, should we try to solve it together in this PR or just create another issue?

@damithc Oops, my bad :(